error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/FileRules.java

```diff
@@ -174,4 +177,70 @@ boolean after(File file) {
       return file.mkdirs() || file.exists();
     }
   }
+
+  /**
+   * Prefer {@link Files#newBufferedReader(Path)} over creating a {@link BufferedReader} from a
+   * {@link FileInputStream} for a string path.
+   */
+  static final class FilesNewBufferedReaderFromStringPath {
+    @BeforeTemplate
+    @SuppressWarnings("DefaultCharset" /* This violation will be rewritten. */)
+    BufferedReader before(String path) throws IOException {
+      return new BufferedReader(new InputStreamReader(new FileInputStream(path)));
+    }
+
+    @AfterTemplate
+    BufferedReader after(String path) throws IOException {
+      return Files.newBufferedReader(Path.of(path));
+    }
+  }
```

Comment:
Here and below: strictly speaking this isn't behavior-preserving, as we change the charset used. I'll add a comment calling this out.

(I also considered adding an extra template to also rewrite the UTF-8 case, but i.c.w. the suggestion above that case is taken care of when chained with the one below.)